File: JavaBasics/JavaOOP/Geometry/Shapes/PlaneShapes/Triangle.java
package Geometry.Shapes.PlaneShapes;

import Geometry.Vertices.Vertex;
import com.sun.javaws.exceptions.InvalidArgumentException;

import java.util.ArrayList;

public class Triangle extends PlaneShape {

    private Vertex pointA;
    private Vertex pointB;
    private Vertex pointC;
    private double sideA;
    private double sideB;
    private double sideC;
    private ArrayList<Vertex> vertices;

    public Triangle(Vertex pointA, Vertex pointB, Vertex pointC) throws InvalidArgumentException {
        super(pointA);

        this.pointA = super.getVertex();
        this.setPointB(pointB);
        this.setPointC(pointC);

        this.sideA = pointC.calculateDistance(pointB);
        this.sideB = pointC.calculateDistance(pointA);
        this.sideC = pointA.calculateDistance(pointB);

        this.isValidTriangleCheck();

        this.vertices = super.getVertices();
        vertices.add(pointB);
        vertices.add(pointC);
    }

    private void isValidTriangleCheck() throws InvalidArgumentException {
        boolean isValidA = sideB + sideC > sideA;
        boolean isValidB = sideA + sideC > sideB;
        boolean isValidC = sideA + sideB > sideC;
        boolean isValidTriangle = isValidA && isValidB && isValidC;
        if (!isValidTriangle) {
            String[] message = new String[]{"Invalid triangle points."};
            throw new InvalidArgumentException(message);
        }
    }

    public Vertex getPointB() {
        return this.pointB;
    }

    public void setPointB(Vertex pointB) {
        this.pointB = pointB;
    }

    public Vertex getPointC() {
        return this.pointC;
    }

    public void setPointC(Vertex pointC) {
        this.pointC = pointC;
    }

    @Override
    public double getArea() {
        double halfPerimeter = this.getPerimeter() / 2;
        double area = Math.sqrt(
                halfPerimeter * (halfPerimeter - this.sideA) * (halfPerimeter - this.sideB) * (halfPerimeter - this.sideC));

        return area;
    }

    @Override
    public double getPerimeter() {
        double perimeter = this.sideA + this.sideB + sideC;
        return perimeter;
    }

    @Override
    public String toString() {
        StringBuilder triangle = new StringBuilder();
        triangle.append(String.format(
                "Triangle:%na = %.2f;%nb = %.2f;%nc = %.2f;%nPerimeter = %.2f;%nArea = %.2f;%n",
                this.sideA, this.sideB, this.sideC, this.getPerimeter(), this.getArea()));
        triangle.append(String.format("Coordinates: %s%n", this.vertices.toString()));

        return triangle.toString();
    }
}

File: C#/ProgrammingBasics/Homework/ConditionalStatements/Triangle/triangle.cs
using System;

    class triangle
    {
        static void Main(string[] args)
        {
            int aX = int.Parse(Console.ReadLine());
            int aY = int.Parse(Console.ReadLine());
            int bX = int.Parse(Console.ReadLine());
            int bY = int.Parse(Console.ReadLine());
            int cX = int.Parse(Console.ReadLine());
            int cY = int.Parse(Console.ReadLine());

            double a = Math.Sqrt(Math.Pow(bX - cX, 2) + Math.Pow(bY - cY, 2));
            double b = Math.Sqrt(Math.Pow(aX - cX, 2) + Math.Pow(aY - cY, 2));
            double c = Math.Sqrt(Math.Pow(bX - aX, 2) + Math.Pow(bY - aY, 2));

            bool isTriangle = (a + b > c) && (a + c > b) && (c + b > a);

            double p = (a + b + c) / 2;

            if (isTriangle)
            {
                Console.WriteLine("Yes");
                double area = Math.Sqrt(p * (p - a) * (p - b) * (p - c));
                Console.WriteLine("{0:F2}", area);
            }
            else
            {
                Console.WriteLine("No");
                Console.WriteLine("{0:F2}", c);
            }
        }
    }

